bindings/capacitorjs/android/src/main/java/org/iota/IotaWalletMobile/IotaWalletMobile.java

```diff
@@ -0,0 +1,190 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+package org.iota.IotaWalletMobile;
+
+import android.util.Log;
+
+import com.getcapacitor.JSArray;
+import com.getcapacitor.JSObject;
+import com.getcapacitor.Plugin;
+import com.getcapacitor.PluginCall;
+import com.getcapacitor.PluginMethod;
+import com.getcapacitor.annotation.CapacitorPlugin;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.iota.Wallet;
+import org.iota.api.WalletCommand;
+import org.iota.types.ClientConfig;
+import org.iota.types.CoinType;
+import org.iota.types.WalletConfig;
+import org.iota.types.events.Event;
+import org.iota.types.events.EventListener;
+import org.iota.types.events.transaction.TransactionProgressEvent;
+import org.iota.types.events.wallet.WalletEventType;
+import org.iota.types.exceptions.WalletException;
+import org.iota.types.secret.StrongholdSecretManager;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import static org.iota.api.NativeApi.callBaseApi;
+import static org.iota.api.NativeApi.callListen;
+
+
+@CapacitorPlugin(name = "IotaWalletMobile")
+public class IotaWalletMobile extends Plugin {
+
+    Wallet wallet = null;
+
+    @PluginMethod()
+    public void messageHandlerNew(final PluginCall call) throws JSONException {
+
+        if (!call.getData().has("clientOptions")
+                || !call.getData().has("storagePath")
+                || !call.getData().has("coinType")
+                || !call.getData().has("secretManager")) {
+            call.reject("clientOptions, storagePath, coinType and secretManager are required");
+        }
+        JSObject _clientOptions = call.getObject("clientOptions");
+        String _storagePath = call.getString("storagePath");
+        Integer _coinType = call.getInt("coinType");
+        JSObject _secretManager = call.getObject("secretManager");
+        String path = getContext().getFilesDir() + _storagePath;
+        Log.d("_clientOptions", _clientOptions.get("nodes").toString());
+
+
+        JSONArray _nodes = _clientOptions.getJSONArray("nodes");
+        String[] nodes = new String[_nodes.length()];
+        for (int i = 0; i < _nodes.length(); i++) {
+            JSONObject _node = (JSONObject) _nodes.get(i);
+            nodes[i] = _node.get("url").toString();
+        }
```

Comment:
Is this needed for all arrays? We have more than just `nodes`

Reply:
Right, to be implemented with this task https://github.com/iotaledger/wallet.rs/issues/1961